apps/src/lib/util/firehose.js

```diff
@@ -209,9 +213,18 @@ class FirehoseClient {
    */
   putRecord(
     data,
-    options = {alwaysPut: false, includeUserId: false, callback: null}
+    options = {
+      alwaysPut: false,
+      includeUserId: false,
+      callback: null,
+      useProgressScriptId: true
```

Comment:
why does this default to true?

Reply:
I think because this was the default behavior assumed in many other calls to putRecord.

Reply:
yes, to what Dave said below.  My weird customization is to make useProgressScriptId false so that script_id is set manually, but most of the time we want it to be set based on progress  so the default for useProgressScriptId is true.

Reply:
oh, right, that makes sense!